sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -199,10 +215,32 @@ export class SenderImpl implements Sender {
     return this._isClosed || this._context.isClosed;
   }
 
-  async send(message: ServiceBusMessage, options?: OperationOptions): Promise<void> {
+  async send(message: ServiceBusMessage, options?: OperationOptions): Promise<void>;
+  async send(messages: ServiceBusMessage[], options?: OperationOptions): Promise<void>;
+  async send(
+    messageOrMessages: ServiceBusMessage | ServiceBusMessage[],
+    options?: OperationOptions
+  ): Promise<void> {
     this._throwIfSenderOrConnectionClosed();
-    throwTypeErrorIfParameterMissing(this._context.namespace.connectionId, "message", message);
-    return this._sender.send(message, options);
+
+    if (Array.isArray(messageOrMessages)) {
+      const batch = await this.createBatch(options);
+
+      for (const message of messageOrMessages) {
+        // we'll let the service throw it's normal error rather than
+        // attempt to do that here.
+        batch.tryAdd(message);
```

Comment:
If this message results in batch getting bigger, then `tryAdd()` doesnt really add the message to the batch. So, you wont get the service error.

You have a few options:
- Dont use the `ServiceBusMessageBatch` and do what we did in Track 1 i.e manually batch up all the messages into a single AMQP message and send it. The service will throw error if too big
- Use the `ServiceBusMessageBatch` and throw an error here when `tryAdd()` returns false. This is what you had initially. My [feedback](https://github.com/Azure/azure-sdk-for-js/pull/8570#pullrequestreview-401416822) was to use the same error type/code as one would get if they had sent a big message to service instead of making a custom error.

Reply:
Option #2 seems best, in light of the fixes that Chris is planning.

Reply:
Done.